src/main/java/org/opensearch/geospatial/ip2geo/listener/Ip2GeoListener.java

```diff
@@ -0,0 +1,118 @@
+    private final ThreadPool threadPool;
+
+    private final DatasourceFacade datasourceFacade;
+
+    @Inject
+    public Ip2GeoListener(final ClusterService clusterService, final ThreadPool threadPool, final DatasourceFacade datasourceFacade) {
+        this.clusterService = clusterService;
+        this.threadPool = threadPool;
+        this.datasourceFacade = datasourceFacade;
+    }
+
+    @Override
+    public void clusterChanged(final ClusterChangedEvent event) {
+        if (event.localNodeClusterManager() == false) {
+            return;
+        }
+
+        for (RestoreInProgress.Entry entry : event.state().custom(RestoreInProgress.TYPE, RestoreInProgress.EMPTY)) {
+            if (RestoreInProgress.State.SUCCESS.equals(entry.state()) == false) {
+                continue;
+            }
+
+            if (entry.indices().stream().anyMatch(index -> DatasourceExtension.JOB_INDEX_NAME.equals(index)) == false) {
+                continue;
+            }
+
+            threadPool.generic().submit(() -> forceUpdateGeoIpData());
+        }
+    }
+
+    private void forceUpdateGeoIpData() {
+        datasourceFacade.getAllDatasources(new ActionListener<>() {
+            @Override
+            public void onResponse(final List<Datasource> datasources) {
+                datasources.stream().forEach(Ip2GeoListener.this::scheduleForceUpdate);
+                datasourceFacade.updateDatasource(datasources, new ActionListener<>() {
+                    @Override
+                    public void onResponse(final BulkResponse bulkItemResponses) {
+                        log.info("Datasources are updated for cleanup");
+                    }
+
+                    @Override
+                    public void onFailure(final Exception e) {
+                        log.error("Failed to update datasource for cleanup after restoring", e);
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(final Exception e) {
+                log.error("Failed to get datasource after restoring", e);
+            }
+        });
+    }
+
+    /**
+     *  Give a delay so that job scheduler can schedule the job right after the delay. Otherwise, it schedules
+     *  the job after specified update interval.
+     */
+    private void scheduleForceUpdate(Datasource datasource) {
+        IntervalSchedule schedule = new IntervalSchedule(Instant.now(), SCHEDULE_IN_MIN, ChronoUnit.MINUTES, DELAY_IN_MILLIS);
+        datasource.resetDatabase();
+        datasource.setSystemSchedule(schedule);
+        datasource.setTask(DatasourceTask.ALL);
+    }
+
+    @Override
+    protected void doStart() {
+        if (DiscoveryNode.isClusterManagerNode(clusterService.getSettings())) {
+            clusterService.addListener(this);
+        }
+    }
+
+    @Override
+    protected void doStop() {
+        clusterService.removeListener(this);
+    }
+
+    @Override
+    protected void doClose() throws IOException {
+
```

Comment:
should we do same as on stop here?

Reply:
I think `doStop` is called always before `doClose()`. Removing listener in `doClose()` will have no effect.
Saw the same pattern in OpenSearch core.